src/Arrays/MPIStateArrays.jl

```diff
@@ -281,9 +304,17 @@ Base.setindex!(Q::MPIStateArray, x...; kw...) =
 
 Base.eltype(Q::MPIStateArray, x...; kw...) = eltype(Q.data, x...; kw...)
 
-Base.Array(Q::MPIStateArray) = Array(Q.data)
+Base.Array(Q::MPIDestArray) = Array(rawview(Q))
```

Comment:
I think we are overusing the Union `MPIDestArray`. It should only be necessary for the LHS of a broadcast `copyto!`

Reply:
Is there any down-side for doing it this way, though? I mean, why not generalize this to an MPIDestArray?

I could imagine people just wanting to transfer an MPIDestArray to the CPU and doing it in this way.